src/app/mailviewer/rmm7messageactions.ts

```diff
@@ -35,6 +36,15 @@ export class RMM7MessageActions implements MessageActions {
     draftDeskService: DraftDeskService;
     rmmapi: RunboxWebmailAPI;
 
+    public updateMessages(messageIds: number[],
+                          updateIndex: (messageIds: number[]) => void,
+                          updateAPI: (messageIds: number[]) => Observable<any>,
+                         ): Observable<any> {
+        updateIndex(messageIds);
+        updateAPI(messageIds).subscribe(() => this.searchService.updateIndexWithNewChanges());
+        return of(null);
```

Comment:
Any reason why it returns an observable? It's not dependent on the result of previous, async actions, so in practice the function is as good as if it was a synchronous one, I think. If we wanted it to be a "notify me when finished" thing, we'd have to pipe off of updateAPI or so.

I like the overall idea, but I'd reconsider Observables here – Promises might be more fitting, since they actually execute their actions even if you don't subscribe to them, and also don't have a habit of executing twice if you subscribe to them twice :P

Reply:
The idea was that after all this was actually finished, the calling func can do things like "remove selections" etc

Reply:
.. right, realised it didnt actually need to do that at all (at least not in this case) .. updated! now to review the others..